### PR TITLE
Update usnic BTL to libfabric v1.4

### PR DESCRIPTION
**Pull request by jsquyres**
Do not merge until https://github.com/ofiwg/libfabric/issues/2052 is closed.

### CODE REVIEW

**Comment by jsquyres** [comment]
@bturrubiates Please review.

File: opal/mca/btl/usnic/btl_usnic_component.c [hunk 1]
```diff
@@ -590,6 +592,30 @@ static void free_filter(usnic_if_filter_t *filter)
...
+
+    hints.caps = FI_MSG;
+    hints.mode = FI_LOCAL_MR | FI_MSG_PREFIX;
+    hints.addr_format = FI_SOCKADDR;
+    hints.ep_attr = &ep_attr;
+    hints.fabric_attr = &fabric_attr;
+
+    int ret;
+    struct fi_info *info_list_local;
+    ret = fi_getinfo(version, NULL, 0, 0, &hints, &info_list_local);
```
Comment (bturrubiates) [inline opal/mca/btl/usnic/btl_usnic_component.c]: why not:

```c
return fi_getinfo(version, NULL, 0, 0, &hints, *info_list_local);
```

and get rid of `info_list_local` and `ret`?

File: opal/mca/btl/usnic/btl_usnic_component.c [hunk 2]
```diff
@@ -841,6 +881,8 @@ static mca_btl_base_module_t** usnic_component_init(int* num_btl_modules,
         module->fabric = fabric;
         module->domain = domain;
         module->fabric_info = info;
+        module->linux_device_name = strdup(linux_device_name);
```
Comment (bturrubiates) [inline opal/mca/btl/usnic/btl_usnic_component.c]: Shouldn't you be checking the return value of `strdup` here?

Comment (jsquyres) [inline opal/mca/btl/usnic/btl_usnic_component.c]: Yes, I should.  Good catch.

File: opal/mca/btl/usnic/btl_usnic_component.c (hunk 1 above)
Comment (jsquyres) [inline opal/mca/btl/usnic/btl_usnic_component.c]: I think you meant

```c
return fi_getinfo(version, NULL, 0, 0, &hints, info_list);
```

But yes, you're right -- that's a bit simpler.

**Comment by jsquyres** [comment]
@bturrubiates Pushed new commits addressing your comments.  Still waiting for Travis to clear CI.

**Comment by bturrubiates** [comment]
👍

**Pull request merged**